### PR TITLE
Update experimental API reference docs to include IS and OD

### PR DESCRIPTION
**Pull request by consoull**
### Linked issue(s):
Fixes KOL-6559

### What change does this PR introduce and why?
Updates experimental API reference docs to include instance_segmentation and object_detection packages.

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/_experimental/instance_segmentation/dataset.py [hunk 1]
```diff
@@ -11,26 +11,54 @@
...
+from typing import Dict
+from typing import Literal
+from typing import Union
+
+import pandas as pd
+
 from kolena._experimental.object_detection.dataset import upload_object_detection_results
 
 
-upload_instance_segmentation_results = upload_object_detection_results
```
Comment (consoull) [inline kolena/_experimental/instance_segmentation/dataset.py:17]: Unfortunately this was the only way I could get mkdocstring to pick up the alias

File: kolena/_experimental/instance_segmentation/dataset.py [hunk 2]
```diff
@@ -11,26 +11,54 @@
...
+    min_confidence_score: float = 0.01,
+    batch_size: int = 10_000,
+) -> None:
+    """
+    Compute metrics and upload results of the model for the dataset.
 
-Dataframe `df` should include a `locator` column that would match to that of corresponding datapoint. Column
-:inference in the Dataframe `df` should be a list of scored [`Polygons`][kolena.annotation.Polygon].
+    Dataframe `df` should include a `locator` column that would match to that of corresponding datapoint. Column
+    :inference in the Dataframe `df` should be a list of scored [`Polygons`][kolena.annotation.Polygon].
```
Comment (munkyshi) [inline kolena/_experimental/instance_segmentation/dataset.py]: Not new to this change, but it looks like the formatting for `:inference` isn't working as intended -- not sure if there's a quick fix or not.

**Review by munkyshi — APPROVED** [review]
I noticed that we seem to have lost the special formatting for functions (might just be a local thing for me):

Prod:
https://docs.kolena.com/reference/initialize/
Screenshot 2024-05-28 at 15 36 11

Local:
http://localhost:8999/reference/initialize/
Screenshot 2024-05-28 at 15 36 02

Is that known / expected? I'm getting this on `1.19.0` as well, so I assume it isn't a regression, but it is harder to read API docs.

File: kolena/_experimental/instance_segmentation/dataset.py (hunk 2 above)
Comment (consoull) [reply kolena/_experimental/instance_segmentation/dataset.py]: Good catch. I updated the docstring to match how `locator` column is referenced prior

**Pull request merged**